sys/can/conn/raw.c

```diff
@@ -0,0 +1,280 @@
+
+    mbox_init(&conn->mbox, conn->mbox_queue, CONN_CAN_RAW_MBOX_SIZE);
+    conn->flags = flags;
+    conn->count = 0;
+    conn->ifnum = ifnum;
+
+    if (flags & CONN_CAN_RECVONLY) {
+        can_opt_t opt;
+        opt.opt = CANOPT_LISTEN_ONLY;
+        opt.data = NULL;
+        opt.data_len = 0;
+        int ret = raw_can_set_can_opt(ifnum, &opt);
+        if (ret < 0) {
+            return ret;
+        }
+    }
+
+    return conn_can_raw_set_filter(conn, filter, count);
+}
+
+
+int conn_can_raw_set_filter(conn_can_raw_t *conn, struct can_filter *filter, size_t count)
+{
+    assert(conn != NULL);
+    assert(filter != NULL || count == 0);
+
+    DEBUG("conn_can_raw_set_filter: conn=%p, filter=%p, count=%d\n",
+          (void *)conn, (void *)filter, count);
+    DEBUG("conn_can_raw_set_filter: conn->filter=%p, conn->count=%d\n",
+          (void *)conn->filter, conn->count);
+
+    /* unset previous filters */
+    if (conn->count) {
+        for (size_t i = 0; i < conn->count; i++) {
+            DEBUG("conn_can_raw_set_filter: unsetting filter=0x%" PRIx32 ", mask=0x%" PRIx32 "\n",
+                 conn->filter[i].can_id, conn->filter[i].can_mask);
+            raw_can_unsubscribe_rx_mbox(conn->ifnum, &conn->filter[i], &conn->mbox, conn);
+        }
+    }
+
+    for (size_t i = 0; i < count; i++) {
+        DEBUG("conn_can_raw_set_filter: setting filter=0x%" PRIx32 ", mask=0x%" PRIx32 "\n",
+              filter[i].can_id, filter[i].can_mask);
+        int ret = raw_can_subscribe_rx_mbox(conn->ifnum, &filter[i], &conn->mbox, conn);
+        if (ret < 0) {
+            DEBUG("conn_can_raw_set_filter: error setting filters %d\n", ret);
+            for (size_t j = 0; j < i; j++) {
+                DEBUG("conn_can_raw_set_filter: unsetting filter=0x%" PRIx32 ", mask=0x%" PRIx32 "\n",
+                      filter[j].can_id, filter[j].can_mask);
+                raw_can_unsubscribe_rx_mbox(conn->ifnum, &filter[j], &conn->mbox, conn);
+            }
+            return ret;
+        }
+    }
+
+    conn->filter = filter;
+    conn->count = count;
+
+    return 0;
+}
+
+#ifdef MODULE_XTIMER
+static void _tx_conf_timeout(void *arg)
+{
+    conn_can_raw_t *conn = arg;
+    msg_t msg;
+
+    msg.type = _TIMEOUT_TX_MSG_TYPE;
+    msg.content.value = _TIMEOUT_MSG_VALUE;
+
+    mbox_put(&conn->mbox, &msg);
+}
+#endif
+
+int conn_can_raw_send(conn_can_raw_t *conn, const struct can_frame *frame, int flags)
+{
+    assert(conn != NULL);
+    assert(conn->ifnum < CAN_DLL_NUMOF);
+    assert((conn->flags & CONN_CAN_RECVONLY) == 0);
+    assert(frame != NULL);
+
+    int ret = 0;
+    int handle;
+
+    DEBUG("conn_can_raw_send: conn=%p, frame=%p, flags=%d\n",
+          (void *)conn, (void *)frame, flags);
+
+    if (flags & CONN_CAN_DONTWAIT) {
+        handle = ret = raw_can_send(conn->ifnum, frame, 0);
+        if (ret >= 0) {
+            ret = 0;
+        }
+    }
+    else {
+#ifdef MODULE_XTIMER
```

Comment:
Just curious: do you have any specific use case in mind where you want to disable xtimer?